src/Workspaces/Core/MSBuild.BuildHost/BuildHost.cs

```diff
@@ -43,36 +43,61 @@ private bool TryEnsureMSBuildLoaded(string projectOrSolutionFilePath)
                 return true;
             }
 
-            VisualStudioInstance? instance;
+            if (!PlatformInformation.IsRunningOnMono)
+            {
+
+                VisualStudioInstance? instance;
 
 #if NETFRAMEWORK
 
-            // In this case, we're just going to pick the highest VS install on the machine, in case the projects are using some newer
-            // MSBuild features. Since we don't have something like a global.json we can't really know what the minimum version is.
+                // In this case, we're just going to pick the highest VS install on the machine, in case the projects are using some newer
+                // MSBuild features. Since we don't have something like a global.json we can't really know what the minimum version is.
 
-            // TODO: we should also check that the managed tools are actually installed
-            instance = MSBuildLocator.QueryVisualStudioInstances().OrderByDescending(vs => vs.Version).FirstOrDefault();
+                // TODO: we should also check that the managed tools are actually installed
+                instance = MSBuildLocator.QueryVisualStudioInstances().OrderByDescending(vs => vs.Version).FirstOrDefault();
 
 #else
 
-            // Locate the right SDK for this particular project; MSBuildLocator ensures in this case the first one is the preferred one.
-            // TODO: we should pick the appropriate instance back in the main process and just use the one chosen here.
-            var options = new VisualStudioInstanceQueryOptions { DiscoveryTypes = DiscoveryType.DotNetSdk, WorkingDirectory = Path.GetDirectoryName(projectOrSolutionFilePath) };
-            instance = MSBuildLocator.QueryVisualStudioInstances(options).FirstOrDefault();
+                // Locate the right SDK for this particular project; MSBuildLocator ensures in this case the first one is the preferred one.
+                // TODO: we should pick the appropriate instance back in the main process and just use the one chosen here.
+                var options = new VisualStudioInstanceQueryOptions { DiscoveryTypes = DiscoveryType.DotNetSdk, WorkingDirectory = Path.GetDirectoryName(projectOrSolutionFilePath) };
+                instance = MSBuildLocator.QueryVisualStudioInstances(options).FirstOrDefault();
 
 #endif
 
-            if (instance != null)
-            {
-                MSBuildLocator.RegisterInstance(instance);
-                _logger.LogInformation($"Registered MSBuild instance at {instance.MSBuildPath}");
-                return true;
+                if (instance != null)
+                {
+                    MSBuildLocator.RegisterInstance(instance);
+                    _logger.LogInformation($"Registered MSBuild instance at {instance.MSBuildPath}");
+                }
+                else
+                {
+                    _logger.LogCritical("No compatible MSBuild instance could be found.");
```

Comment:
I kind of preferred how it was before where we would explicitly return false in all the failure cases.  IMHO made it a bit easier to read.

Reply:
I got too scared that we'd have a place where we returned true but forgot to call register or vice versa.